### DISCUSSION TITLE
How to Access DLL Project's Own Assets with AssetLoader in Avalonia?

### DISCUSSION THREAD

**Discussion by xushushun** [AvaloniaUI/Avalonia#17673] [Q&A]
I have a solution with two projects: a main application project and a dynamic plugin project (DLL). The main program uses Activator.CreateInstance to load the DLL plugin. In the DLL, I can access the main program’s assets using the avares: scheme. However, the DLL project also has its own assets, but I can't access them using AssetLoader.Open(new Uri("avares:...")) in the DLL code.

Why am I unable to access the DLL's own assets this way, and how can I resolve this

**Comment by timunie** [comment]
You should be able to, just make sure it's included as AvaloniaResource and you are using full path including dll name

**Reply by xushushun** [reply]
I've identified the cause of the issue. I need to reference the DLL project in the main project during compilation in order to access the DLL's own assets. This means that every time I update the program, I must recompile both the main and DLL projects, which defeats the purpose of making the DLL modular. Is there any way to solve this?

**Reply by xushushun** [reply]
> You should be able to, just make sure it's included as AvaloniaResource and you are using full path including dll name

If I directly reference the DLL in a non-plugin mode, there is no problem. However, when loading the DLL using the Activator.CreateInstance method, there is an issue.
